Src/Sarif.PatternMatcher/JsonLogicalLocationProcessor.cs

```diff
@@ -38,7 +39,24 @@ public void Process(ICollection<Result> results, string fileContents)
                 {
                     string lastPath = null;
 
+                    fileRegionsCache ??= new FileRegionsCache();
+
+                    var sortedByLocation = new SortedDictionary<int, Result>();
+
                     foreach (Result result in results)
+                    {
+                        Location location = result.Locations[0];
+                        Region region = location.PhysicalLocation.Region;
+
+                        fileRegionsCache.PopulateTextRegionProperties(region,
```

Comment:
We might run some additional computations, but I doubt it'd be measurable. Determining that some properties haven't, in fact, been set can be tricky in our system, because we don't insist on nullable numeric values everywhere. We would therefore need to start looking for clues like do we have a start line, etc.

All things considered, I'm not sure that complexity is worth it. In SPAM, we never actually hit these code paths: we will always provide comprehensive properties. And so from that angle, the right response is probably to drop this additional logic: we don't use it and we aren't necessarily interested in a public API for this scenario.